OpenMP backend fails to compile with clang-17

When compiling the unit tests with clang-17, enabling the `ACC_CPU_B_SEQ_T_OMP2` backend, the following failure is reported:
```
/home/bgruber/dev/alpaka/include/alpaka/atomic/AtomicOmpBuiltIn.hpp:190:50: error: the statement for 'atomic compare capture' must be a compound statement of form '{v = x; cond-up-stmt}', ''{cond-up-stmt v = x;}', '{if(x == e) {x = d;} else {v = x;}}', '{r = x == e; if(r) {x = d;}}', or '{r = x == e; if(r) {x = d;} else {v = x;}}', where 'cond-update-stmt' can have one of the following forms: 'if(expr ordop x) {x = expr;}', 'if(x ordop expr) {x = expr;}', 'if(x == e) {x = d;}', or 'if(e == x) {x = d;}' where 'x', 'r', and 'v' are lvalue expressions with scalar type, 'expr', 'e', and 'd' are expressions with scalar type, and 'ordop' is one of '<' or '>'.
  190 |                     ref = (ref <= value) ? ref : value;
      |                                                  ^~~~~
```

--- a/include/alpaka/atomic/AtomicOmpBuiltIn.hpp
+++ b/include/alpaka/atomic/AtomicOmpBuiltIn.hpp
@@ -187,7 +187,8 @@ namespace alpaka
 #        pragma omp atomic capture compare
                 {
                     old = ref;
-                    ref = (ref <= value) ? ref : value;
+                    if(value < ref)
+                        ref = value;
                 }
                 return old;
             }
@@ -205,7 +206,8 @@ namespace alpaka
 #        pragma omp atomic capture compare
                 {
                     old = ref;
-                    ref = (ref >= value) ? ref : value;
+                    if(value > ref)
+                        ref = value;
                 }
                 return old;
             }
@@ -217,21 +219,12 @@ namespace alpaka
         {
             ALPAKA_FN_HOST static auto atomicOp(AtomicOmpBuiltIn const&, T* const addr, T const& value) -> T
             {
+                // TODO(bgruber): atomic increment with wrap around is not implementable in OpenMP 5.1
                 T old;
-                auto& ref(*addr);
-// atomically update ref, but capture the original value in old
-#        if BOOST_COMP_GNUC
-#            pragma GCC diagnostic push
-#            pragma GCC diagnostic ignored "-Wconversion"
-#        endif
-#        pragma omp atomic capture compare
+#        pragma omp critical(AlpakaOmpAtomicOp)
                 {
-                    old = ref;
-                    ref = ((ref >= value) ? 0 : (ref + 1));
+                    old = AtomicInc{}(addr, value);
                 }
-#        if BOOST_COMP_GNUC
-#            pragma GCC diagnostic pop
-#        endif
                 return old;
             }
         };
@@ -242,21 +235,12 @@ namespace alpaka
         {
             ALPAKA_FN_HOST static auto atomicOp(AtomicOmpBuiltIn const&, T* const addr, T const& value) -> T
             {
+                // TODO(bgruber): atomic decrement with wrap around is not implementable in OpenMP 5.1
                 T old;
-                auto& ref(*addr);
-// atomically update ref, but capture the original value in old
-#        if BOOST_COMP_GNUC
-#            pragma GCC diagnostic push
-#            pragma GCC diagnostic ignored "-Wconversion"
-#        endif
-#        pragma omp atomic capture compare
+#        pragma omp critical(AlpakaOmpAtomicOp)
                 {
-                    old = ref;
-                    ref = ((ref == 0) || (ref > value)) ? value : (ref - 1);
+                    old = AtomicDec{}(addr, value);
                 }
-#        if BOOST_COMP_GNUC
-#            pragma GCC diagnostic pop
-#        endif
                 return old;
             }
         };
@@ -293,8 +277,8 @@ namespace alpaka
             ALPAKA_FN_HOST static auto atomicOp(AtomicOmpBuiltIn const&, T* const addr, T const& value) -> T
             {
                 T old;
-// \TODO: Currently not only the access to the same memory location is protected by a mutex but all atomic ops on all
-// threads.
+                // \TODO: Currently not only the access to the same memory location is protected by a mutex but all
+                // atomic ops on all threads.
 #        pragma omp critical(AlpakaOmpAtomicOp)
                 {
                     old = TOp()(addr, value);
@@ -309,8 +293,8 @@ namespace alpaka
                 T const& value) -> T
             {
                 T old;
-// \TODO: Currently not only the access to the same memory location is protected by a mutex but all atomic ops on all
-// threads.
+                // \TODO: Currently not only the access to the same memory location is protected by a mutex but all
+                // atomic ops on all threads.
 #        pragma omp critical(AlpakaOmpAtomicOp2)
                 {
                     old = TOp()(addr, compare, value);

